src/_arraykit.c

```diff
@@ -147,6 +148,11 @@ AK_ResolveDTypeIter(PyObject *dtypes)
         }
     }
     Py_DECREF(iterator);
+    if (!resolved) {
+        // this could happen if this function gets an empty tuple
+        PyErr_Format(PyExc_ValueError, "Iterator passed to resolve dtypes is empty.");
+        return NULL;
```

Comment:
Original code is fine... just saw opportunities for a bit of cleanup:

```suggestion
        PyErr_SetString(PyExc_ValueError, "iterator passed to resolve dtypes is empty");
```